test/model/tools/model_test/main.cpp

```diff
@@ -441,6 +444,30 @@ reduce_counterexample(std::shared_ptr<model::kv_workload> workload, const std::s
 {
     reduce_counterexample_context_t context{main_home, reduce_home, conn_config, table_config};
 
+    /*
+     * Turn off generating core dumps during the counterexample reduction. Each failed run could
+     * possibly result in dumping a core, leading to tens or even hundreds of core dumps.
+     */
+    struct rlimit prev_core_limit;
+    int r = getrlimit(RLIMIT_CORE, &prev_core_limit);
+    if (r != 0)
+        throw std::runtime_error(
+          std::string("Failed to get the current core dump limit: ") + strerror(errno));
+
+    model::at_cleanup reset_core_limit([&prev_core_limit]() {
+        int r = setrlimit(RLIMIT_CORE, &prev_core_limit);
+        if (r != 0)
+            std::cerr << "Failed to reset the core dump limit: " << strerror(errno) << std::endl;
+    });
+
+    struct rlimit new_core_limit;
+    memset(&new_core_limit, 0, sizeof(new_core_limit));
+    /* Preserve the maximum limit for the resetting the current value to work at cleanup. */
+    new_core_limit.rlim_max = prev_core_limit.rlim_max;
```

Comment:
Sorry I am not sure why we use `prev_core_limit.rlim_max`, why don't we set the limits to 0?
```suggestion
    new_core_limit.rlim_max = 0;
```

Reply:
To turn off core dumps, we just need to set `rlim_cur` to 0, which is done by the `memset` above.

If we set `rlim_max` to 0, then calling `setrlimit(RLIMIT_CORE, &prev_core_limit)` would fail with `EPERM`, because we tried to set `rlim_cur` to a value that's larger than the current `rlim_max`. (I wonder if this would require two calls to `setrlimit`, one to increase the maximum and one to increase the current setting, if that's even allowed?)

Reply:
But the call `setrlimit(RLIMIT_CORE, &prev_core_limit)` should work as we are not altering the struct `prev_core_limit`.

Reply:
This was discussed offline - the code is good this way, thank you!